src/test/java/com/google/edith/ReceiptFileHandlerServiceTest.java

```diff
@@ -0,0 +1,130 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.edith;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.FileInfo;
+import com.google.appengine.tools.development.testing.LocalBlobstoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.edith.services.ReceiptFileHandlerService;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class ReceiptFileHandlerServiceTest {
+
+  private final LocalServiceTestHelper testHelper =
+      new LocalServiceTestHelper(
+          new LocalBlobstoreServiceTestConfig(), new LocalDatastoreServiceTestConfig());
+
+  private ReceiptFileHandlerService receiptFileHandlerService;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    testHelper.setUp();
+    receiptFileHandlerService = new ReceiptFileHandlerService(blobstoreService);
+  }
+
+  @After
+  public void tearDown() {
+    testHelper.tearDown();
+  }
+
+  @Mock BlobstoreService blobstoreService;
+
+  @Mock HttpServletRequest request;
+
+  @Mock HttpServletResponse response;
+
+  @Test
+  public void checks_ifBlobDidNotUpload_returnsNull() throws IOException {
+    Map<String, List<FileInfo>> emptyMap = Collections.emptyMap();
+
+    when(blobstoreService.getFileInfos(request)).thenReturn(emptyMap);
+    Optional<List<FileInfo>> uploadedFileInfo =
+        receiptFileHandlerService.getUploadedFileUrl(request, "example");
+    assertFalse(uploadedFileInfo.isPresent());
+  }
+
+  /** Checks if a file uploaded successfully in Blobstore returns BlobInfo. */
+  @Test
+  public void checks_ifBlobUploaded_returnsFileInfo() throws IOException {
+    Date creationDate = new Date();
+    List<FileInfo> files = new ArrayList<FileInfo>();
```

Comment:
I think we can use ImmutableList here and other tests to get in the habit of using immutable in tests and source code

List<FileInfo> files = ImmutableList.of();

Reply:
Thanks for the suggestion. Changes made.